javascript/tables/src/readTable.utils.ts

```diff
@@ -39,21 +39,20 @@ export function getParser(options?: ParserOptions) {
 function arqueroParser(text: string, options?: ParserOptions): ColumnTable {
 	const mappedOptions = mapProps(ParserType.Arquero, options) as CSVParseOptions
 	let table = fromCSV(text, mappedOptions)
-	if (options?.nRows) {
-		const subset = nRows([...table], options.nRows)
-		table = from(subset)
-	}
-	return table
+	return options?.readRows ? from([...table].slice(0, options.readRows)) : table
 }
 
-function papaParser(text: string, options?: ParserOptions): ParseResult<any> {
+function papaParser(text: string, options?: ParserOptions): ColumnTable {
+	if (options?.skipRows && options?.readRows) {
+		options.readRows += options.skipRows
```

Comment:
I wouldn't mutate the actual input param. It's much safer to clone it first like `const opts = {...options}`. This may also result in an empty object if options is not present, which would remove the need for the `?` guards.